### ISSUE TITLE
Certain inline html not rendering correctly

### ISSUE THREAD

**Issue by fperez** [executablebooks/jupyterlab-mystjs#31]
### Describe the bug

**context**

I have notebooks with inline html that isn't rendering correctly.

### Reproduce the bug

In a markdown cell, the following content:

```
Use the <button class='btn btn-default btn-xs'><i class='icon-stop fa fa-stop'></i></button> button in the toolbar above.
```

renders in "normal" JupyterLab correctly as:

![image](https://user-images.githubusercontent.com/57394/217385706-721f7c9b-12eb-4aa3-b5eb-11d6ab0ec37e.png)

while mystjs renders it incorrectly as:

![image](https://user-images.githubusercontent.com/57394/217385805-9be99431-96c5-454a-a661-add74ea96c46.png)

### List your environment

```
(stat159-s23) longs[~]> jupyter-book --version
Jupyter Book      : 0.13.1
External ToC      : 0.2.4
MyST-Parser       : 0.15.2
MyST-NB           : 0.13.2
Sphinx Book Theme : 0.3.3
Jupyter-Cache     : 0.4.3
NbClient          : 0.5.13
```

**Comment by fperez** [comment]
This is with mystjs 0.1.3 btw.

**Comment by agoose77** [comment]
@rowanc1 based upon the fact that the https://myst.spec renderer shows the (seemingly) correct AST but non-rendered output, I think this is a `mystjs` bug rather than something specific to this repo?

**Comment by rowanc1** [comment]
Yep, this is going to be a deeper issue in mystjs parser. It was originally all aimed towards secure html / semantic meaning, so we disabled html (both a security hole & difficult to translate to all the various other renderers). @fwkoch is doing surgery in the parser at the moment, and this would be a good thing to consider as well if we can!

**Comment by rowanc1** [comment]
We fixed up certain pieces in `v0.1.4`, but it is likely that the `<button>/<i>` listed in this issue are not going to work.

Currently HTML is actually being parsed and trying to be "understood", rather than fully rendered as is. We are doing this because MyST likes to understand more for rendering out to other formats like Word/LaTeX/JATS. I think that the behaviour in JupyterLab need to mimic a bit closer what happens today, and that is going to be a different chunk of work!

Leaving this open for now, but it should have some improvements in `v0.1.4`!

**Comment by mcauthorn** [comment]
One more quick update on this, which I suspect may be useful for those looking for a work around (granted, it's not ideal):
 If I coerce the rendering by converting it to a code cell and use the IPython magic command:
```
%%html
<div>
<img src="/files/Jupyter-Logo-DemOS.png" width="250"/>
</div>
```
The HTML renders as expected.

**Comment by rowanc1** [comment]
Another work around in that case is to use MyST, an image or figure directive! :)
https://myst-tools.org/docs/mystjs/figures#image-directive

````md
```{figure} https://source.unsplash.com/random/400x200?beach,ocean
:name: myFigure
:alt: Random image of the beach or ocean!
:align: center

Relaxing at the beach 🏝 🌊 😎
```
````
